packages/python/javdb_platform/db.py

```
@@ -1818,21 +2552,432 @@ def db_create_report_session(
+                            )
+                except Exception as e:
+                    counts['drift_skipped'] += 1
+                    logger.error(
+                        "Rollback step failed (table=%s action=%s id=%s): %s",
+                        main_table, action, target_id, e,
+                    )
+
+            if counts['drift_skipped'] == drift_before:
+                # Tidy up the audit rows we just consumed.
+                conn.execute(
+                    f"DELETE FROM {audit_table} WHERE SessionId=?", (session_id,),
```

Ruff (0.15.12)

[error] Possible SQL injection vector through string-based query construction

(S608)